### ISSUE TITLE
Focus not following the mouse [bug]

### ISSUE THREAD

**Issue by bobwilkinson20** [noah-nuebling/Mac-Mouse-Fix#701]
## Description

Just installed version 2.2.3 on Ventura 13.6.1. I was able to remap buttons 4 & 5 successfully so it seems to be working. The problem is that focus is no longer automatically following the mouse. For example when I open slack and then put the cursor in the channel list on left I can still use the mouse (scrolling works, can click), but the scroll bar doesn't show until I actually click somewhere in that part of the window. Another example - in chrome when I hover the pointer over the 'x' to close a tab it no longer shows the focus there.

## Steps to reproduce

see Description.

## Context

Mac Mouse Fix Version: 2.2.3
macOS Version: Ventura 13.6.1
Mouse Model: Microsoft MAIN-35956 Sculpt Ergonomic Mouse

## Additional Comments

–

---

### Console Logs

–

### Crash Reports

–

**Comment by bobwilkinson20** [comment]
another symptom. hover the pointer near the edge of the window to resize it. The cursor does not change to the resize cursor.

**Comment by bobwilkinson20** [comment]
this is a major issue and my ability to use my mac is significantly degraded. there is other behavior that is broken now too. miro is unusable now - for some reason it's not registering any mouse button clicks. 

I have uninstalled mac mouse fix and rebooted my machine and things are still broken. Please help understand what this software does at installation time and how I can undo it.

**Comment by noah-nuebling** [comment]
Hi there @bobwilkinson20, thanks for reaching out! This was never a feature in Mac Mouse Fix. I think [CursorSense](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwit7eHh0KiCAxWj2AIHHW0PDxkQFnoECBAQAQ&url=https%3A%2F%2Fplentycom.jp%2Fen%2Fcursorsense%2F&usg=AOvVaw28SexLPxvIzrbMYMXJXiHd&opi=89978449) can do it IIRC. I hope that helps a little!

**Comment by noah-nuebling** [comment]
Sorry I didn't read your messages fully. Let me try again: Any changes in cursor behavior that differ from default, native macOS behaviour can't be caused by Mac Mouse Fix, as far as I know, so I assume that you must have some other mouse software installed which is affecting the cursor behaviour. It might be SteerMouse or CursorSense. Those apps provide similar features to what you describe as far as I know. If you're trying to use Mac Mouse Fix at the same time as these apps, that might lead to interferences and problems, which I can't solve easily.

If you try to uninstall any other mouse apps aside from Mac Mouse Fix or - alternatively - disable Mac Mouse Fix, that should resolve the interferences which might fix your issues. 

If you uncheck the "Enable Mac Mouse Fix" checkbox, then Mac Mouse Fix should not have any effect on your mouse's behaviour whatsoever. If you want to be absolutely sure that Mac Mouse Fix is not affecting your mouse, you can check for a process called "Mac Mouse Fix Helper" in Activity Monitor. Mac Mouse Fix can only have any effect on your mouse if "Mac Mouse Fix Helper" is running. This process should disappear when you disable Mac Mouse Fix.

I hope this is more helpful than my last response, and let me know if you have more questions.

**Comment by bobwilkinson20** [comment]
The issues were definitely caused by Mac Mouse Fix. I haven't installed any other mouse/pointer/cursor software at any point. Just for context - the reason I was attempting MMF is that my Microsoft Mouse developed an issue where the thumb button was hypersensitive. That button default to a 'back' operation which was dropping me out of video calls on a regular basis. I went looking for a way to remap that button and MMF is what I found. 

I did a bit more troubleshooting since opening this bug...

First, even after doing an uninstall using AppCleaner and restarting, the problem persisted.

Second, the problem is specific to having the Microsoft mouse connected to my machine. It was connected to my machine via a USB-C hub. With the mouse connected, the touchpad had all of the same problematic behavior. When I disconnected the USC-C hub, though, the cursor/pointer via the touchpad returned to normal operation.

Third, I then tried a logitech mouse (also via USB-C hub). The cursor/pointer all worked normally with that Mouse.

I can still reproduce the problems at will by connecting the MSFT mouse. You will have much better understanding than I do, but these symptoms are highly suggestive that MMF changed something on my machine that is specific to the MSFT mouse as they showed up exactly at the point I installed MMF and disappear when I do not have that mouse connected to my machine.

**Comment by noah-nuebling** [comment]
I'm sorry I don't think I can be of much help here, but I'll try. As I said, from what I know about Mac Mouse Fix, it only can have any effect on your mouse if the "Mac Mouse Fix Helper" process is running. There are no system configurations that are changed by Mac Mouse Fix which could lead to any effects that persist after "Mac Mouse Fix Helper" is not running anymore. If you uninstalled the app, emptied the bin, and restarted the computer, then Mac Mouse Fix can't have any effect on your mouse based on my understanding. 

There might be some way that Mac Mouse Fix could have effects after you've uninstalled it wich I've forgotten about. But since I've implemented the program, I think that's highly unlikely. 

I know you've ruled this out, but the only explanation I can think of is that there is some other utility installed on your system causing these issues. I believe this especially since focus following the mouse is not a native macOS behavior as far as I'm aware of. I did some Googling and I found this tool called [AutoRaise](https://github.com/sbmpost/AutoRaise) which enables focus following the mouse. My only idea is that you might have such a tool running in the background causing these weird behaviours.

You mentioned that your Microsoft mouse had the behaviour of going back and forward when pressing the side buttons. This is also not a native macOS feature. As far as I'm aware, you need to install third party software to get back and forward behaviour from your mouse buttons, except if your mouse is sending keyboard shortuts when you press the buttons. But if the buttons were sending keyboard shortcuts, then Mac Mouse Fix couldn't recognize those buttons. So since you mentioned that Mac Mouse Fix did recognize the buttons, I think that this also points to you having some other mouse driver installed which was responsible for the back and forward behaviour of your mouse buttons.

I hope that's a little bit helpful and that you can fix your problem soon.
